### PR TITLE
LPS-168568 Generalize logic in dm display context

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/browse/LPS-168568

There's some unnecessary logic in display contexts to filter out system structures and hide some controls.

Infer that information from the file entry type or ddm structure and remove any custom logic.

# How do I test this?

There should be no changes in behavior. You can verify the following:

- Create a video shortcut or google docs shortcut and verify that the structure fileds are not displayed in the info panel, and that the video/document cannot be downloaded.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:help

**Comment by adolfopa** [comment]
ci:reevaluate:1367401_2454

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLViewFileVersionDisplayContext.java [hunk 1]
```diff
@@ -200,7 +205,10 @@ public List<DDMStructure> getDDMStructures() throws PortalException {
 			DLFileVersion dlFileVersion =
 				(DLFileVersion)_fileVersion.getModel();
 
-			_ddmStructures = dlFileVersion.getDDMStructures();
+			_ddmStructures = ListUtil.filter(
+				dlFileVersion.getDDMStructures(),
+				ddmStructure ->
+					ddmStructure.getType() != DDMStructureConstants.TYPE_AUTO);
```
Comment (AliciaGarciaGarcia) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLViewFileVersionDisplayContext.java:211]: When you create a Document type, the type of the ddmStructure is updated to `TYPE_AUTO` , so we are hiding the metadata on the info_panel not only on the google/videos but all

https://github.com/liferay-lima/liferay-portal/blob/47319863a397b61f650554f512657e0f7f6bc34f/modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/DataEngineDLFileEntryTypeLocalServiceWrapper.java#L101

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]
When you create a Document type, the type of the ddmStructure is updated to `TYPE_AUTO` , so we are hiding the metadata on the info_panel not only on the google/videos but all

https://github.com/liferay-lima/liferay-portal/blob/47319863a397b61f650554f512657e0f7f6bc34f/modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/DataEngineDLFileEntryTypeLocalServiceWrapper.java#L101


I do not think this is the wanted behavior

**Comment by adolfopa** [comment]
> When you create a Document type, the type of the ddmStructure is updated to `TYPE_AUTO` , so we are hiding the metadata on the info_panel not only on the google/videos but all

I'll look into this and push-force additional changes.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia All changes squashed into [bc71a9a](https://github.com/liferay-lima/liferay-portal/pull/3730/commits/bc71a9a40ab88bfb29f243453e5eac1b5cf381c6)

**Comment by adolfopa** [comment]
ci:reevaluate:1277401_5842

**Comment by adolfopa** [comment]
ci:test:document

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
waiting for the test:document result

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**